jb-yoda/src/main/java/life/jukebox/YodaSubscriberClient.java

```diff
@@ -0,0 +1,36 @@
+package life.jukebox.yoda;
+
+import com.google.cloud.ServiceOptions;
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.SubscriptionName;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+
+public class YodaSubscriberClient{
+  private static final String PROJECT_ID = getEnvironmentVariable("GCP_PROJECT_ID");
+  private static final String SUBSCRIPTION_ID = getEnvironmentVariable("PUBSUB_SUB_JUKEBOX_CREATED_YODA");
```

Comment:
Did you figure out who would consume the exception in the event of one?

Reply:
Short answer is I still don't know. <clint> is what initiates the call to this method and is assigned by JVM so I'm not really sure how to catch it.

Reply:
Looked into it - init is a constructor for the instance, and non-static field initialization. clinit is for the static initialization blocks for the class, and static field initialization. This means you can't catch it since you don't control clinit execution. 
As I said, you should print the message "Env var .. is not set" to standard error, and exit with a nonzero code.